### PR TITLE
Fix FC/BC layer for EventDispatcher

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Fix https://github.com/lexik/LexikJWTAuthenticationBundle/pull/637

Having Contracts 1.1 doesn't mean EventDispatcher 4.3 is installed ;)

ping @chalasr

### CODE REVIEW

File: Event/Event.php [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 namespace Lexik\Bundle\JWTAuthenticationBundle\Event;
 
-use Symfony\Component\EventDispatcher\Event as BaseEvent;
-use Symfony\Contracts\EventDispatcher\Event as ContractsBaseEvent;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 
-if (class_exists(ContractsBaseEvent::class)) {
+if (is_subclass_of(EventDispatcher::class, EventDispatcherInterface::class) {
```
Comment (tucksaun) [inline Event/Event.php]: missing end `)`

Comment (nicolas-grekas) [reply Event/Event.php]: fixed

**Comment by chalasr** [comment]
Good catch, thanks @nicolas-grekas.

**Comment by chalasr** [comment]
Released in v2.6.3

File: Event/Event.php [hunk 2]
```diff
@@ -3,9 +3,11 @@
 namespace Lexik\Bundle\JWTAuthenticationBundle\Event;
 
 use Symfony\Component\EventDispatcher\Event as BaseEvent;
+use Symfony\Component\EventDispatcher\EventDispatcher;
 use Symfony\Contracts\EventDispatcher\Event as ContractsBaseEvent;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
 
-if (class_exists(ContractsBaseEvent::class)) {
+if (is_subclass_of(EventDispatcher::class, EventDispatcherInterface::class)) {
     class Event extends ContractsBaseEvent
```
Comment (yceruto) [inline Event/Event.php]: Type error in https://github.com/symfony/symfony/blob/fba11b4dc34e7c589d8c30d5b6a090387d52e648/src/Symfony/Component/EventDispatcher/Debug/TraceableEventDispatcher.php#L314

Reported in this repo https://github.com/lexik/LexikJWTAuthenticationBundle/issues/647 though I'm not sure if something can be improved in Symfony.

**Pull request merged**